osu.Game/Screens/Play/HitErrorDisplay/DefaultHitErrorDisplay.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Rulesets.Judgements;
+using osuTK.Graphics;
+using osuTK;
+using osu.Framework.Graphics.Sprites;
+using System.Collections.Generic;
+using osu.Game.Rulesets.Objects;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Colour;
+using osu.Game.Graphics;
+using osu.Framework.Extensions.Color4Extensions;
+using System.Linq;
+
+namespace osu.Game.Screens.Play.HitErrorDisplay
+{
+    public class DefaultHitErrorDisplay : HitErrorDisplay
+    {
+        private const int stored_judgements_amount = 5;
+        private const int bar_width = 3;
```

Comment:
these could be ordered better, and xmldoc'd where it's not obvious what they are (`stored_judgements_amount` and `judgement_life_time` are good candidates).